src/aeso_ast_infer_types.erl

```diff
@@ -2250,6 +2304,13 @@ pp_when({check_expr, Expr, Inferred0, Expected0}) ->
 pp_when({checking_init_type, Ann}) ->
     io_lib:format("when checking that 'init' returns a value of type 'state' at ~s\n",
                   [pp_loc(Ann)]);
+pp_when({list_comp, BindExpr, Inferred0, Expected0}) ->
```

Comment:
Maybe I should consider cases of `comprehension_bind` and `comprehension_if`, @UlfNorell what do you think?